app/web/src/features/dashboard/CommunityBrowser.tsx

```diff
@@ -0,0 +1,196 @@
+        //a previously unselected item is clicked
+        setSelected([...selected.slice(0, Math.max(level, 0)), community]);
+        setCachedQueryResults(cachedQueryResults.slice(0, level + 1));
+      }
+    }
+  };
+
+  return (
+    <div className={classes.root}>
+      {cachedQueryResults.map((query, index) => (
+        <BrowserColumn
+          key={index}
+          parent={selected?.[index - 1] ?? globalCommunityQuery.data}
+          communities={query.data}
+          handleClick={(community) => handleClick(community, index)}
+          selected={selected[index]?.communityId}
+        />
+      ))}
+      {query.isLoading ? ( // div prevents overflow scrollbar from spinner
+        <div className={classes.loader}>
+          <CircularProgress />
+        </div>
+      ) : query.isSuccess && globalCommunityQuery.isSuccess ? (
+        <div ref={lastColumnRef}>
+          <BrowserColumn
+            parent={
+              selected?.[selected.length - 1] ?? globalCommunityQuery.data
+            }
+            communities={query.data.pages.flatMap(
+              (page) => page.communitiesList
+            )}
+            handleClick={(community) => handleClick(community, selected.length)}
+          />
+          {query.hasNextPage && (
+            <Button
+              onClick={() => query.fetchNextPage()}
+              loading={query.isFetchingNextPage}
+              variant="outlined"
+            >
+              {LOAD_MORE}
+            </Button>
+          )}
+        </div>
+      ) : (
+        <Alert severity="error">{query?.error?.message || ""}</Alert>
+      )}
+    </div>
+  );
+}
+
+function BrowserColumn({
+  parent,
+  communities,
+  handleClick,
+  selected,
+}: {
+  parent?: Community.AsObject;
+  communities: Community.AsObject[];
+  handleClick: (community: Community.AsObject) => void;
+  selected?: number;
+}) {
+  const classes = useStyles();
+
+  return (
+    <List className={classes.list}>
+      {parent && (
+        <>
+          <ListItem
+            component={StyledLink}
+            to={routeToCommunity(parent.communityId, parent.slug)}
+          >
+            {parent.name}
+          </ListItem>
+          <Divider />
+        </>
+      )}
+      {communities.map((community) => (
+        <ListItem
+          key={community.communityId}
+          button
+          onClick={() => handleClick(community)}
+          aria-selected={community.communityId === selected}
+        >
+          <ListItemText
+            primaryTypographyProps={{
+              className:
+                community.communityId === selected
+                  ? classes.selected
+                  : undefined,
+            }}
+          >
+            {community.name}
+          </ListItemText>
+        </ListItem>
+      ))}
+    </List>
+  );
+}
+
+const StyledLink = forwardRef<
+  HTMLAnchorElement,
+  { to: string; children: ReactNode }
+>((props, ref) => <MuiLink component={Link} {...props} ref={ref} />);
```

Comment:
Might it be worth pulling this into `components`? Seems like something we could re-use wherever we need a React Router link?